modules/lang-painless/src/test/java/org/elasticsearch/painless/WhenThingsGoWrongTests.java

```diff
@@ -197,4 +201,13 @@ public void testDynamicListWrongIndex() {
             exec("def x = new ArrayList(); x.add('foo'); return x['bogus'];");
         });
     }
+
+    public void testRCurlyNotDelim() {
+        ScriptException e = expectThrows(ScriptException.class, () -> {
+            // We don't want PICKY here so we get the normal error message
+            exec("def i = 1} return 1", emptyMap(), emptyMap(), null);
+        });
```

Comment:
Picky has a performance cost and we only enable it in tests (https://github.com/elastic/elasticsearch/blob/master/modules/lang-painless/src/main/java/org/elasticsearch/painless/antlr/Walker.java#L191-L207)

As far as error messages for the user, we don't really test them directly from unit tests. If you want to see what the user will actually get, you have to call .toJsonString() on the ScriptException. I think these kind of integration tests are best done from REST?

Instead today in unit tests we test the actual `cause` of the script exception: `expectScriptThrows`. Some unit tests also assert that the offset (line number) is reasonable too. If these things are good, then the ScriptException to the user will be good.